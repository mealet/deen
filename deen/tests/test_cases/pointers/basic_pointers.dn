fn main() i32 {
  // reference to single stack value
  let a = &10;
  println!("{}", *a);

  *a = 20;
  println!("{}", *a);

  // reference to variable
  let var = 123;
  let var_ptr = &var;
  println!("{} | {}", var, *var_ptr);

  *var_ptr = 100;
  println!("{} | {}", var, *var_ptr);

  // multi-references
  let var = 555;
  let multi_ptr = &&var;

  println!("{} | {}", var, **multi_ptr);
  
  **multi_ptr = 0;

  println!("{} | {}", var, **multi_ptr);

  return 0;
}
