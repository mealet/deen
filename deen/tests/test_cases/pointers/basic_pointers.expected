10
20
123 | 123
100 | 100
555 | 555
0 | 0
